database/src/backend/tests.rs

```diff
@@ -331,6 +322,60 @@ fn db_read_write() {
     }
 }
 
+/// Assert that `key`'s in database tables are sorted in
+/// an ordered B-Tree fashion, i.e. `min_value -> max_value`.
+#[test]
+fn tables_are_sorted() {
```

Comment:
After our dup-key changes, is this trait still required?
```rust
trait Key: Storable {
    fn compare(left: &[u8], right: &[u8]) -> Ordering;
}
```
This test asserts that keys are ordered by value (at least, `u64`).

Reply:
Not needed but if we are going to be getting values using ranges then yes.

Probably something for the future but right now when getting a range of values we do a look up for each one whereas using an iterator we can get them all with one search. It will be hard to make this multi-threaded though

Reply:
Okay, will keep for now.